test/tests/transfers/multiapp_userobject_transfer/tests

```diff
@@ -23,4 +23,12 @@
     exodiff = '3d_1d_master_out.e 3d_1d_master_out_sub_app0.e'
     recover = false
   [../]
+
+  [./3d_1d_err]
+    type = 'RunException'
+    input = '3d_1d_master.i'
+    cli_args = 'Transfers/layered_transfer_from_sub_app/injective_master_to_sub_map=true'
+    expect_err = 'MultiAppUserObjectTransfer: Master element with centroid at'
+    prereq = 3d_1d
```

Comment:
On these new tests, let's add the 'requirement', 'design', and 'issues' parameters. Look at moose/test/tests/postprocessors/num_iterations/tests for an example.

Reply:
I have added these additional parameters to the test files. I have also added documentation for MultiAppUserObjectTransfer and LineValueSampler based on my understanding of the these classes. Please let me know if any changes need to be made to test descriptions or added documentation.